docs/smart-contracts/meebits_erc721.sol

```diff
@@ -0,0 +1,679 @@
+        require(msg.value >= salePrice, "Insufficient funds to purchase.");
+        if (msg.value > salePrice) {
+            msg.sender.transfer(msg.value.sub(salePrice));
+        }
+        beneficiary.transfer(salePrice);
+        numSales++;
+        return _mint(msg.sender, 0);
+    }
+
+    function _mint(address _to, uint createdVia) internal returns (uint) {
+        require(_to != address(0), "Cannot mint to 0x0.");
+        require(numTokens < TOKEN_LIMIT, "Token limit reached.");
+        uint id = randomIndex();
+
+        numTokens = numTokens + 1;
+        _addNFToken(_to, id);
+
+        emit Mint(id, _to, createdVia);
+        emit Transfer(address(0), _to, id);
+        return id;
+    }
+
+    function _addNFToken(address _to, uint256 _tokenId) internal {
+        require(idToOwner[_tokenId] == address(0), "Cannot add, already owned.");
+        idToOwner[_tokenId] = _to;
+
+        ownerToIds[_to].push(_tokenId);
+        idToOwnerIndex[_tokenId] = ownerToIds[_to].length.sub(1);
+    }
+
+    function _removeNFToken(address _from, uint256 _tokenId) internal {
+        require(idToOwner[_tokenId] == _from, "Incorrect owner.");
+        delete idToOwner[_tokenId];
+
+        uint256 tokenToRemoveIndex = idToOwnerIndex[_tokenId];
+        uint256 lastTokenIndex = ownerToIds[_from].length.sub(1);
+
+        if (lastTokenIndex != tokenToRemoveIndex) {
+            uint256 lastToken = ownerToIds[_from][lastTokenIndex];
+            ownerToIds[_from][tokenToRemoveIndex] = lastToken;
+            idToOwnerIndex[lastToken] = tokenToRemoveIndex;
+        }
+
+        ownerToIds[_from].pop();
+    }
+
+    function _getOwnerNFTCount(address _owner) internal view returns (uint256) {
+        return ownerToIds[_owner].length;
+    }
+
+    function _safeTransferFrom(address _from,  address _to,  uint256 _tokenId,  bytes memory _data) private canTransfer(_tokenId) validNFToken(_tokenId) {
+        address tokenOwner = idToOwner[_tokenId];
+        require(tokenOwner == _from, "Incorrect owner.");
+        require(_to != address(0));
+
+        _transfer(_to, _tokenId);
+
+        if (isContract(_to)) {
+            bytes4 retval = ERC721TokenReceiver(_to).onERC721Received(msg.sender, _from, _tokenId, _data);
+            require(retval == MAGIC_ON_ERC721_RECEIVED);
+        }
+    }
+
+    function _clearApproval(uint256 _tokenId) private {
+        if (idToApproval[_tokenId] != address(0)) {
+            delete idToApproval[_tokenId];
+        }
+    }
+
+    //// Enumerable
+
+    function totalSupply() public view returns (uint256) {
+        return numTokens;
+    }
+
+    function tokenByIndex(uint256 index) public pure returns (uint256) {
+        require(index >= 0 && index < TOKEN_LIMIT);
+        return index + 1;
+    }
+
+    function tokenOfOwnerByIndex(address _owner, uint256 _index) external view returns (uint256) {
+        require(_index < ownerToIds[_owner].length);
+        return ownerToIds[_owner][_index];
+    }
```

Comment:
Accepts an owner address and index then returns a Meebit token ID. You can use this by first calling balanceOf on an address to get the number of meebits an address owns. Then you can call tokenOfOwnerByIndex up to the balance total returned by the balanceOf call. So if a address 0x0 has a balanceOf 3 meebits, you can make the following three calls to get the three meebit token IDs for this address:
tokenOfOwnerByIndex("0x0", 0)
tokenOfOwnerByIndex("0x0", 1)
tokenOfOwnerByIndex("0x0", 2)

Reply:
Thanks, wasn't aware of that function. 
WoollyEyedWalnut #3378 on meebits dao discord, created a quick jsfiddle for interacting with the contract and see he has used it
https://jsfiddle.net/walnut2walnut/uyhkoec9/327/